src/cpp/core/system/Win32OutputCapture.cpp

```diff
@@ -44,13 +44,22 @@ void standardStreamCaptureThread(
       {
          const int kBufferSize = 512;
          char buffer[kBufferSize];
-         DWORD bytesRead = 0;
-         if (::ReadFile(hReadPipe, &buffer, kBufferSize, &bytesRead, NULL))
+         // read from the descriptor; this descriptor is attached to a pipe,
+         // and this _read call blocks until we have some bytes or until the
+         // descriptor is closed
+         DWORD bytesRead = ::_read(readFd, &buffer, kBufferSize);
+         if (bytesRead > 0)
          {
-            if (bytesRead > 0)
-               outputHandler(std::string(buffer, bytesRead));
+            // if we got some bytes, invoke the output handler
+            outputHandler(std::string(buffer, bytesRead));
          }
-         else
+         else if (bytesRead == 0)
```

Comment:
AFAIK we should never stop capturing -- do we expect to ever see bytesRead == 0 (perhaps at end of process?). If we don't every expect to see it perhaps we should log a warning here?